src/clib/initialize_metal_chemistry_rates.cpp

```diff
@@ -248,141 +190,134 @@ int grackle::impl::initialize_metal_chemistry_rates(
+    }
+    kcol_rate_tables->data[CollisionalRxnLUT::k150][i] = 3.0e-10 * exp(-6717.0/ttt);
+    if(ttt < 4000.0) {
+      kcol_rate_tables->data[CollisionalRxnLUT::k151][i] = 1.6e-14 * pow(ttt, -0.33);
+    } else {
+      kcol_rate_tables->data[CollisionalRxnLUT::k151][i] = 1.0e-15;
+    }
+    kcol_rate_tables->data[CollisionalRxnLUT::k152][i] = 9.1e-10;
+    kcol_rate_tables->data[CollisionalRxnLUT::k153][i] = 1.7e-7 * pow(ttt, -0.5);
   
-        my_rates->kz15[i] = 4.98e-11;
-        my_rates->kz16[i] = 2.70e-10;
-        my_rates->kz17[i] = 7.00e-14 * pow(ttt300, 2.80) * exp(-1950.0/ttt);
-        my_rates->kz18[i] = 6.83e-12 * pow(ttt300, 1.60) * exp(-9720.0/ttt);
-        my_rates->kz19[i] = 3.30e-10 * exp(-8460.0/ttt);
-        my_rates->kz20[i] = 6.64e-10 * exp(-11700.0/ttt);
-        if(ttt < 1.0e7)
-            my_rates->kz21[i] = 3.43e-13 * pow(ttt300, 2.67) * exp(-3160.0/ttt);
-        else
-            my_rates->kz21[i] = 3.43e-13 * pow(1.0e7/300.0, 2.67) * exp(-3160.0/1.0e7);
-        // The rate comes from an experiment (297-3532 K).
-        // We refrain to extrapolate it to high temperatures.
-        my_rates->kz22[i] = 7.00e-10 * exp(-232.0/ttt);
-        my_rates->kz23[i] = 2.38e-10 * exp(-1760.0/ttt);
-        my_rates->kz24[i] = 1.55e-12 * pow(ttt300, 1.60) * exp(-1660.0/ttt);
-        my_rates->kz25[i] = 1.65e-12 * pow(ttt300, 1.14) * exp(-50.0/ttt);
-        my_rates->kz26[i] = 1.0e-13;
-        my_rates->kz27[i] = 1.0e-17;
-        my_rates->kz28[i] = 1.1e-10 * pow(ttt300, 0.5);
-        my_rates->kz29[i] = 3.3e-11;
-        my_rates->kz30[i] = 9.9e-19 * pow(ttt300, -0.38);
-        my_rates->kz31[i] = 4.9e-20 * pow(ttt300, 1.58);
-        my_rates->kz32[i] = 6.6e-11;
-        my_rates->kz33[i] = 4.34e-11 * pow(ttt300, -0.5) * exp(-30.0/ttt);
-        my_rates->kz34[i] = 2.1e-9;
-        my_rates->kz35[i] = 6.9e-9;
-        my_rates->kz36[i] = 2.0e-9;
-        my_rates->kz37[i] = 7.7e-10;
-        my_rates->kz38[i] = 6.2e-10;
-        my_rates->kz39[i] = 6.8e-10;
-        my_rates->kz40[i] = 1.7e-9;
-        my_rates->kz41[i] = 1.01e-9;
-        my_rates->kz42[i] = 8.3e-10;
-        my_rates->kz43[i] = 7.5e-10;
-        my_rates->kz44[i] = 4.4e-12 * pow(ttt300, -0.61);
-        my_rates->kz45[i] = 3.4e-12 * pow(ttt300, -0.63);
-        my_rates->kz46[i] = 1.6e-7 * pow(ttt300, -0.5);
-        my_rates->kz47[i] = 2.0e-7 * pow(ttt300, -0.5);
-        my_rates->kz48[i] = 3.5e-7 * pow(ttt300, -0.5);
-        my_rates->kz49[i] = 6.5e-7 * pow(ttt300, -0.5);
-        my_rates->kz50[i] = 1.95e-7 * pow(ttt300, -0.7);
-        my_rates->kz51[i] = 1.0e-17;
-        my_rates->kz52[i]  = 3.00e-11;
-        my_rates->kz53[i]  = 1.30e-11 * exp(-111.0/ttt);
-        my_rates->kz54[i]  = 2.00e-13;
-
-//      printf("CHECK %13.5e %13.5e %13.5e %13.5e\n", ttt
-//         , my_rates->k125[i]
-//         , my_rates->k129[i]
-//         , my_rates->k130[i]);
+    kcol_rate_tables->data[CollisionalRxnLUT::kz15][i] = 4.98e-11;
```

Comment:
You mentioned not being a big fan of the `add_reaction_rate` strategy of defining individual functions for each rate. Do you foresee this getting refactored in another way to unify it with how the rest of the reaction rates are initialized? I actually like the individual rate function strategy for the transparency and ability to provide specific, detailed documentation. Either way, I'm curious where you see this going.